crates/astria-composer/src/searcher/executor.rs

```diff
@@ -164,28 +205,43 @@ impl Executor {
                             break Err(e).wrap_err("failed submitting bundle to sequencer");
                         }
                     }
+
+                    block_timer.as_mut().reset(Instant::now() + Duration::from_millis(self.block_time_ms));
+                }
+
+                Some(next_bundle) = future::ready(bundle_factory.next_finished()), if submission_fut.is_terminated() => {
+                    let bundle = next_bundle.pop();
+                    if !bundle.is_empty() {
+                        submission_fut = self.submit_bundle(nonce, bundle);
+                    }
+                }
+
+                // receive new seq_action and bundle it
+                Some(seq_action) = self.serialized_rollup_transactions_rx.recv() => {
+                    let rollup_id = seq_action.rollup_id;
+                    match bundle_factory.try_push(seq_action) {
+                        Ok(()) => {}
+                        Err(BundleFactoryError::SequenceActionTooLarge{size, max_size}) => {
+                            warn!(
+                                rollup_id = %rollup_id,
+                                seq_action_size = size,
+                                max_size = max_size,
+                                "failed to bundle sequence action: too large. sequence action is dropped."
```

Comment:
The `seq_action_size` and `max_size` fields should come out of the display-formatted error. I.e. derive `thiserror::Error` with the same information and don't set the fields here explicitly. Just match the error with `Err(e)` and then `error = %e as &StdError`